exonum/src/storage/db.rs

```diff
@@ -34,9 +38,43 @@ impl Patch {
         Self { changes: HashMap::new() }
     }
 
-    /// Returns amount of changes.
+    /// Returns changes for the given name.
+    fn get_changes(&self, name: &str) -> Option<&Changes> {
+        self.changes.get(name)
+    }
+
+    /// Returns a mutable reference to the changes corresponding to the `name`.
+    fn get_changes_mut(&mut self, name: &str) -> Option<&mut Changes> {
+        self.changes.get_mut(name)
+    }
+
+    /// Creates a changes map for the given name if doesn't exist already.
+    fn create_changes_if_needed(&mut self, name: &str) {
```

Comment:
@DarkEld3r Can you explain pls, why do we need this method?

Reply:
There are three places with the following code:
```rust
if !self.patch.contains_key(name) {
    self.patch.insert(name.to_string(), BTreeMap::new());
}
```
I suppose it is enough to move it into a separate function. :smile:

Reply:
Instead of call `create_changes_if_needed`  and than `get_changes_mut(...).unwrap()` you may want:

```rust
fn get_changes_mut(...) {
     self.changes.entry(name).or_insert_with(BTreeMap::new)...
}
```